src/metadata.js

```diff
@@ -0,0 +1,55 @@
+var gcpMetadata = require('gcp-metadata');
+var constants = require('./constants.js');
+
+// prevent self tracing
+var headers = {};
+headers[constants.TRACE_AGENT_REQUEST_HEADER] = 1;
+
+module.exports = function getMetadata(logger, fields, cb) {
```

Comment:
I really don't like the string based enum ('continue', 'break') that adds strong coupling between tracewriter and this file.

A few suggestions:
1. Make this file return a promise for a single metadata property.
2. Keep the then functions in trace writer.
3. Use try/catch.